service/Dockerfile

```diff
@@ -0,0 +1,45 @@
+# LS docker file
+
+# Tomcat OS base image
+FROM artifacts.bcp.absa.co.za/docker.io/tomcat:9.0.62-jdk8-corretto
+
+LABEL org.opencontainers.image.authors="ABSA OSS"
+LABEL org.opencontainers.image.title="Login service"
+LABEL org.opencontainers.image.base.name="absaoss/login-service:latest"
+
+ARG BUILD_PROXY
+ARG CONFIG=./src/main/resources/example.application.yaml
+
+ENV LS_WAR_PATH=./target/scala-2.12
+# deploy as root
+ENV LS_WAR=ROOT.war
+
+# build via (docker root = project root):
+# docker build -t absaoss/login-service:latest --build-arg BUILD_PROXY=http://my.cool.proxy.here:3128 --build-arg CONFIG=./src/main/resources/my.awesome.local.application.yaml .
+# run via
+# docker run -p 8080:8080 absaoss/login-service:latest
+
+# test via:
+# http://localhost:8080/token/public-key
+
+
+# Copy Spring application properties
+COPY $CONFIG /opt/application.yaml
+
+COPY $LS_WAR_PATH/*.war /usr/local/tomcat/webapps/$LS_WAR
+
+ENV SPRING_CONFIG_ADDITIONAL-LOCATION=/opt/application.yaml
+
+ENV http_proxy=$BUILD_PROXY
+ENV https_proxy=$BUILD_PROXY
+ENV HTTP_PROXY=$BUILD_PROXY
+ENV HTTPS_PROXY=$BUILD_PROXY
+
+# uncomment and add packages you would need to install via yum if any
+#RUN yum -y update
+#RUN yum -y install htop procps
+
+EXPOSE 8080
```

Comment:
Tomcat is known for using the default port 8443 for SSL HTTPS traffic. Shouldn't it be also added here, not just 8080? It's like that also in Unify project, see its Dockerfile.

That brings me to this (also in the Unify project):

```
# A self-seigned certificate for HTTPS
RUN keytool -genkeypair -keyalg RSA -alias tomcat -keysize 2048 \
    -dname "CN=*.ctodataengprod.aws.dsarena.com, OU=ursa, O=absa, L=Johannesburg, ST=Gauteng, C=za" \
    -validity 365 -storepass changeit -keystore /usr/local/tomcat/conf/selfsigned.p12 -storetype PKCS12
```

So this is for HTTPS - or do you want to deal with this detail later / it's not needed here (yet?)?

Reply:
I have done exactly that, but optionally. so with docker build, one can either use
 - `--build-arg SSL=false` (same as until now - just HTTP @ 8080), or
 - `--build-arg SSL=true --build-arg SSL_DNAME="CN=*.abc.mydomain.com, OU=myproject, O=mycompany, L=Johannesburg, ST=Gauteng, C=za"` to get also HTTPS @ 8443.